lib/src/action.dart

```diff
@@ -14,8 +14,6 @@
 
 import 'dart:async';
 
-typedef void OnData(dynamic event);
```

Comment:
I think it should be possible to type this code a little better based on keeping this typedef and making it generic:

typedef dynamic OnData<T>(T event);

Reply:
Done.